upb/refcounted.c

```diff
@@ -726,6 +726,18 @@ static void freeobj(upb_refcounted *o) {
 bool upb_refcounted_init(upb_refcounted *r,
                          const struct upb_refcounted_vtbl *vtbl,
                          const void *owner) {
+#ifndef NDEBUG
+  // Endianness check.  This is unrelated to upb_refcounted, it's just a
+  // convenient place to put the check that we can be assured will run for
+  // basically every program using upb.
+  const int x = 1;
+#ifdef UPB_BIG_ENDIAN
+  assert(*(char*)&x != 1);
+#else
+  assert(*(char*)&x == 1);
```

Comment:
Sure I can add a note. Though we do waaaay more expensive runtime checks than this when NDEBUG isn't enabled. I am kind of impressed that you are out-performance-paranoia'ing me. :)

Actually if this is truly optimized away, I'm tempted to leave it in for release builds too. The only problem is that I'm not sure what it ought to do if the check fails. It is such a shame that this can't be a compile-time assert!